TFEngine.py

```diff
@@ -1427,8 +1473,25 @@ def extra_fetches_callback(seq_tags, **extra_fetches_out):
     if output_per_seq_file:
       print('Write eval results per seq to %r' % output_per_seq_file, file=log.v3)
       from Util import betterRepr
+
+      # write fetches to file
       with open(output_per_seq_file, 'w') as f:
-        f.write(betterRepr(results_per_seq) + '\n')
+        def create_output_string(seq_tag):
+          """
+          Creates a colon (;) separated output string in the order given by `requested_outputs`
+          :param str seq_tag:
+          """
+          value_list = [results_per_seq[seq_tag][req_out] for req_out in output_per_seq_format]
+          value_list = [' '.join(map(str, v)) if isinstance(v, numpy.ndarray) else str(v) for v in value_list]
+          assert all([all([c not in "\n;" for c in v]) for v in value_list])
+          return ';'.join(value_list)
+
+        corpus_seq_idx_to_idx = {dataset.get_corpus_seq_idx(idx):idx for idx in range(len(results_per_seq))}
```

Comment:
There is a space missing before `idx`. (I think PyCharm should hint about this?)